Sources/Sentry/SentryTraceState.m

```diff
@@ -0,0 +1,160 @@
+#import "SentryTraceState.h"
+#import "SentryDsn.h"
+#import "SentryLog.h"
+#import "SentryOptions+Private.h"
+#import "SentryScope+Private.h"
+#import "SentrySerialization.h"
+#import "SentryTracer.h"
+#import "SentryUser.h"
+
+@implementation SentryTraceStateUser
+
+- (instancetype)initWithUserId:(nullable NSString *)userId segment:(nullable NSString *)segment
+{
+    if (self = [super init]) {
+        _userId = userId;
+        _segment = segment;
+    }
+    return self;
+}
+
+- (instancetype)initWithUser:(nullable SentryUser *)user
+{
+    NSString *segment;
+    if ([user.data[@"segment"] isKindOfClass:[NSString class]]) {
+        segment = user.data[@"segment"];
+    }
+
+    return [self initWithUserId:user.userId segment:segment];
+}
+
+@end
+
+@implementation SentryTraceState
+
+- (instancetype)initWithTraceId:(SentryId *)traceId
+                      publicKey:(NSString *)publicKey
+                    releaseName:(nullable NSString *)releaseName
+                    environment:(nullable NSString *)environment
+                    transaction:(nullable NSString *)transaction
+                           user:(nullable SentryTraceStateUser *)user
+{
+    if (self = [super init]) {
+        if (traceId == nil || publicKey == nil) {
+            NSLog(@"ops");
```

Comment:
Can we use the sentry logger here instead with a proper message?

Reply:
This is not supposed to be here.